File: app/src/main/java/com/example/app_expenses/Transactions_List.java
package com.example.app_expenses;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.baoyz.swipemenulistview.SwipeMenuListView;
import com.example.app_expenses.databinding.FragmentGalleryBinding;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class Transactions_List extends Fragment {

    //private GalleryViewModel galleryViewModel;
    private FragmentGalleryBinding binding;
    private FloatingActionButton add_exp_button;
    static boolean deleted_transList;
    TextView tot_bal, exp_bal;
    ArrayList<String> new_items_big;
    ArrayAdapter<String> adapter_big;
    ListView trans_big;
    HomeFragment home_frag = new HomeFragment();

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        //galleryViewModel = new ViewModelProvider(this).get(GalleryViewModel.class);

        binding = FragmentGalleryBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        ((MainActivity) getActivity()).setActionBarTitle("Transaction List");

        //trans_big = (ListView)root.findViewById(R.id.transaction_list);

        // Set amount to 0
        Add_Transaction.amount_flt = 0;
        Add_Budget.budget_amount = 0;

        // List view setup
        SwipeMenuListView listView = (SwipeMenuListView) root.findViewById(R.id.listView);

        // Display the list of transactions
        new_items_big = Add_Transaction.getItems();
        adapter_big = new ArrayAdapter<>(root.getContext(), android.R.layout.simple_list_item_1, new_items_big);
        listView.setAdapter(adapter_big) ;

        // Add transaction button
        add_exp_button = root.findViewById(R.id.plus_button_list);
        add_exp_button.show();
        add_exp_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openAdd_transaction();
            }
        });

        // Delete Transactions
        //swipeMenuCreated(root, listView, "TransList");

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    public void openAdd_transaction(){
        Intent intent = new Intent(Transactions_List.this.getActivity(), Add_Transaction.class);
        startActivity(intent);
    }

    public void openHome(){
        Intent intent = new Intent(Transactions_List.this.getActivity(), MainActivity.class);
        startActivity(intent);
    }

}

File: app/src/main/java/com/example/app_expenses/Transactions_List.kt
package com.example.app_expenses

import com.google.android.material.textfield.TextInputLayout
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.example.app_expenses.Add_Transaction
import android.widget.TextView
import com.example.app_expenses.R
import com.example.app_expenses.HomeFragment
import com.google.android.material.snackbar.Snackbar
import com.example.app_expenses.Add_Budget
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import com.example.app_expenses.Transactions_List
import com.google.android.material.floatingactionbutton.FloatingActionButton
import androidx.recyclerview.widget.RecyclerView
import android.widget.ArrayAdapter
import android.graphics.drawable.ColorDrawable
import android.content.DialogInterface
import android.view.View
import androidx.navigation.ui.AppBarConfiguration
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView
import androidx.navigation.NavController
import androidx.navigation.ui.NavigationUI
import com.example.app_expenses.activities.MainActivity
import com.example.app_expenses.databinding.FragmentGalleryBinding
import java.util.ArrayList

class Transactions_List : Fragment() {
    private var binding: FragmentGalleryBinding? = null
    private var add_exp_button: FloatingActionButton? = null
    var new_items_big: ArrayList<String>? = null
    var adapter_big: ArrayAdapter<String>? = null
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentGalleryBinding.inflate(inflater, container, false)
        val root: View = binding!!.root

        // Set amount to 0
        Add_Transaction.amount = 0.0
        Add_Budget.budgAmount = 0.0


        // Display the list of transactions
        new_items_big = Add_Transaction.items
        adapter_big = ArrayAdapter(root.context, android.R.layout.simple_list_item_1, new_items_big!!)

        // Add transaction button
        add_exp_button = root.findViewById(R.id.plus_button_list)
        add_exp_button?.show()
        add_exp_button?.setOnClickListener(View.OnClickListener { openAdd_transaction() })

        // Delete Transactions
        //swipeMenuCreated(root, listView, "TransList");
        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    fun openAdd_transaction() {
        val intent = Intent(this@Transactions_List.activity, Add_Transaction::class.java)
        startActivity(intent)
    }

    fun openHome() {
        val intent = Intent(this@Transactions_List.activity, MainActivity::class.java)
        startActivity(intent)
    }
}